### PR TITLE
Caldav.pm: add test_sched_busytime_query

### PR DESCRIPTION
**Pull request by ksmurchison**
This tests a busytime query via a POST to Outbox.

This also tests case-insensitive check of "mailto" in CUAs when PUTting a scheduling resource

### CODE REVIEW

File: Cassandane/Cyrus/Caldav.pm [hunk 1]
```diff
@@ -4594,4 +4595,52 @@ EOF
...
+ORGANIZER:MAILTO:cassandane\@example.com
+END:VEVENT
+END:VCALENDAR
+EOF
+
+    xlog $self, "freebusy query";
+    my $res = $CalDAV->Request('POST', 'Outbox',
+                               $query, 'Content-Type' => 'text/calendar');
+    my $text = Dumper($res);
+    $self->assert_matches(qr/schedule-response/, $text);
```
[the remark this answers is not in the document]
Comment (ksmurchison) [reply Cassandane/Cyrus/Caldav.pm:4643]: Eventually, yes.  But initially I'm just testing that Cyrus didn't return a 403.

[the remark this answers is not in the document]
Comment (rsto) [reply Cassandane/Cyrus/Caldav.pm:4643]: By eventually you mean: sometime later? Or before this patch is merged? My only concern is if that test is the only test for busyquery, then we will not detect any regressions if the query code turns out to return wrong results later.

[the remark this answers is not in the document]
Comment (ksmurchison) [reply Cassandane/Cyrus/Caldav.pm:4643]: We can certainly add to it later to make sure the the return busy data is correct.

**Review by rsto — APPROVED** [review]
LGTM as a basic test for busyquery

**Comment by elliefm** [comment]
Travis thinks this test fails for 3.2, though it's marked as :min_version_3_0.  I haven't looked closely yet but will sooner or later.

In the meantime, off the top of your head, do you think it's found a bug in 3.2?  Or is the ":min_version_3_0" a typo?

**Comment by ksmurchison** [comment]
The min_3_0 is there because I started with a copy of another test and gave it no thought. 
That said, I would think this test should be compatible with 3.2

**Comment by elliefm** [comment]
Thanks, I'll do some investigation later and either bump the min version or backport a fix or something, depending on what I find

**Pull request merged**